### PR TITLE
fix: Update parse log

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: crates/utils/src/script_log.rs [hunk 1]
```diff
@@ -36,18 +39,45 @@ pub enum GwLog {
...
+        if to_address_byte_size == 0 {
+            8
+        } else {
+            28
+        }
+    };
 
-    let mut to_address = [0u8; 20];
-    to_address.copy_from_slice(&data[21..41]);
+    let to_address = match RegistryAddress::from_slice(&data[start..end]) {
```
Comment (blckngm) [inline crates/utils/src/script_log.rs]: Use `get(start..end)` to avoid panic?

File: crates/utils/src/script_log.rs [hunk 2]
```diff
@@ -36,18 +39,45 @@ pub enum GwLog {
     },
 }
 
-fn parse_sudt_log_data(data: &[u8]) -> ([u8; 20], [u8; 20], u128) {
-    assert_eq!(data[0], 20);
-    let mut from_address = [0u8; 20];
-    from_address.copy_from_slice(&data[1..21]);
+fn parse_sudt_log_data(data: &[u8]) -> Result<(RegistryAddress, RegistryAddress, U256)> {
+    let mut start = 0;
```
Comment (jjyr) [inline crates/utils/src/script_log.rs:45]: check length of data to avoid panic.

**Review by jjyr — CHANGES_REQUESTED** [review]

File: crates/utils/src/script_log.rs (hunk 1 above)
Comment (classicalliu) [reply crates/utils/src/script_log.rs]: Updated

File: crates/utils/src/script_log.rs (hunk 2 above)
Comment (classicalliu) [reply crates/utils/src/script_log.rs:45]: Updated

**Review by jjyr — APPROVED** [review]

**Review by blckngm — APPROVED** [review]

**Pull request merged**